src/SFML/Window/WindowBase.cpp

```diff
@@ -146,22 +147,36 @@ bool WindowBase::isOpen() const
 
 
 ////////////////////////////////////////////////////////////
-Event WindowBase::pollEvent()
+std::optional<Event> WindowBase::pollEvent()
 {
-    Event event;
-    if (m_impl && (event = m_impl->popEvent(false)))
-        filterEvent(event);
+    if (m_impl == nullptr)
+        return std::nullopt;
+
+    std::optional<sf::Event> event = m_impl->popEvent(false /* non-blocking */);
+    if (event.has_value())
+        filterEvent(*event);
+
     return event;
 }
 
 
 ////////////////////////////////////////////////////////////
 Event WindowBase::waitEvent()
 {
-    Event event;
-    if (m_impl && (event = m_impl->popEvent(true)))
-        filterEvent(event);
-    return event;
+    if (m_impl == nullptr)
+    {
+        err() << "Fatal error: `waitEvent` invoked on destroyed window -- did you forget a `break` after "
+                 "`window.close()` in your event handling loop?"
+              << std::endl;
+
+        std::abort();
```

Comment:
@ChrisThrasher: alternatively here we could just return another artificial `sf::Event::Closed`, but I'd rather promote the good practice of breaking after closing the window.

Reply:
@ChrisThrasher: I decided to do that in the end, it's much friendlier for the user compared to termination.